mlflow/projects/__init__.py

```diff
@@ -251,46 +264,127 @@ def _maybe_create_conda_env(conda_env_path):
-        command=command, work_dir=work_dir, env_map=env_map, stream_output=stream_output)
-    return SubmittedRun(active_run, pollable_run)
+    from mlflow.projects.submitted_run import LocalSubmittedRun
+    command = _load_project(work_dir, uri)
+    mlflow_run_arr = ["mlflow", "run", work_dir, "-e", entry_point, "-m", MLFLOW_ENTRY_POINT_MODE]
+    if storage_dir is not None:
+        mlflow_run_arr.extend(["--storage-dir", storage_dir])
+    if not use_conda:
+        mlflow_run_arr.append("--no-conda")
+    if run_id:
+        mlflow_run_arr.extend(["--run-id", run_id])
+    if parameters:
+        for key, value in parameters.items():
+            mlflow_run_arr.extend(["-P", "%s=%s" % (key, value)])
+    final_env = os.environ.copy()
+    final_env.update(env_map)
+    if stream_output:
+        popen = subprocess.Popen(
+            mlflow_run_arr, cwd=work_dir, env=final_env, universal_newlines=True)
+    else:
+        popen = subprocess.Popen(
+            mlflow_run_arr, cwd=work_dir, env=final_env, universal_newlines=True,
+            stderr=open(os.devnull, "w"), stdout=open(os.devnull, "w"))
+    return LocalSubmittedRun(run_id, popen, command)
 
 
-def _run_project(project, entry_point, work_dir, parameters, use_conda, storage_dir,
-                 experiment_id, block):
-    """Locally run a project that has been checked out in `work_dir`."""
+def _maybe_set_run_terminated(active_run, status):
+    """
+    If the passed-in active run is defined and still running (i.e. hasn't already been terminated
+    within user code), mark it as terminated with the passed-in status.
+    """
+    if active_run and not RunStatus.is_terminated(active_run.get_run().info.status):
+        active_run.set_terminated(status)
+
+
+def _run_local_noenv(
+        uri, entry_point, use_conda, parameters, storage_dir):
+    project = _load_project(work_dir=uri, uri=uri)
     storage_dir_for_run = _get_storage_dir(storage_dir)
     eprint("=== Created directory %s for downloading remote URIs passed to arguments of "
            "type 'path' ===" % storage_dir_for_run)
-    # Try to build the command first in case the user mis-specified parameters
-    run_project_command = project.get_entry_point(entry_point)\
-        .compute_command(parameters, storage_dir_for_run)
     commands = []
     if use_conda:
-        conda_env_path = os.path.abspath(os.path.join(work_dir, project.conda_env))
+        conda_env_path = os.path.abspath(os.path.join(uri, project.conda_env))
         _maybe_create_conda_env(conda_env_path)
         commands.append("source activate %s" % _get_conda_env_name(conda_env_path))
+    commands.append(
+        project.get_entry_point(entry_point).compute_command(parameters, storage_dir_for_run))
+    _run_entry_point_command(" && ".join(commands))
+
+
+def _run_entry_point_command(command):
+    """
+    Meant to be run in a subprocess via a CLI command triggered in _run_project. Runs an entry point
```

Comment:
This isn't meant to be run in a subprocess, it starts a subprocess, right? Also is the tracking server stuff still true?